Public/Src/Cache/ContentStore/Library/FileSystem/PassThroughFileSystem.cs

```diff
@@ -229,6 +229,7 @@ public void DeleteDirectory(AbsolutePath path, DeleteOptions deleteOptions)
             catch (UnauthorizedAccessException accessException)
             {
                 if ((deleteOptions & DeleteOptions.ReadOnly) != 0 &&
+                    accessException.HResult > 0 &&

```

Comment:
AccessDenied is greater than 0, so why do we need to have an explicit check for > 0?

Reply:
Locally, I was seeing an overflow exception in the following cast to `uint`. I don't know how `HResult` became negative.